### PR TITLE
Create Model for user, game and piece

### PR DESCRIPTION
**Pull request by Linda091**
Create user, game, and piece models based on Bryan's database structure. A bit confused by the belongs_to white & black_player. And by the boolean :color. 

Added devise gem.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :games #need to determine how to tie association to two foreign keys.
```
Comment (ilyakrasnov) [inline app/models/user.rb]: this should be sth like:
```
belongs_to :white_player, class_name: 'User'
belongs_to :black_player, class_name: 'User'
```

File: app/models/game.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Game < ApplicationRecord
+
+  has_many :pieces
+
+  belongs_to :white_player, :class_name => 'User', foreign_key: 'user_id'
```
Comment (ilyakrasnov) [inline app/models/game.rb]: no need for `foreign_key` here and on the next line.. rails will derive it from the table `User`

File: db/migrate/20180510230425_create_pieces.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreatePieces < ActiveRecord::Migration[5.0]
+  def change
+    create_table :pieces do |t|
+      t.integer :game_id
+      t.integer :location_x
+      t.integer :location_y
+      t.string :picture
+      t.boolean :color #not sure about this one. look up boolean color
```
Comment (ilyakrasnov) [inline db/migrate/20180510230425_create_pieces.rb]: I'd suggest naming that field sth like `black` or `white` - then the `boolean` value would make more sense :)

File: db/migrate/20180510221714_create_games.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateGames < ActiveRecord::Migration[5.0]
+  def change
+    create_table :games do |t|
+      t.integer :game_id
```
Comment (ilyakrasnov) [inline db/migrate/20180510221714_create_games.rb]: no need for `game_id` field on the `Game` itself.. by default all table will have an `id`.. we use `game_id` as a reference on pieces for instance - where we store the id of the game the piece belongs to...

Comment (ilyakrasnov) [reply db/migrate/20180510221714_create_games.rb]: you'd need to add to 'white_player_id' and `black_player_id` fields in order to make our associations work

**Comment by Linda091** [comment]
Hi all - I reviewed and resolved the conflicts. Made sure the master branch has my changes to the model & migration files and also left some of Arit's data, i.e. some gems, validations etc.

**Comment by Linda091** [comment]
Updated the model and migration files based on Bryan's database and recs Ilya made. Resolved all conflicts - Srit and I had some gem dupes so corrected tha.  Also confirmed the schema looks like what I currently have on my branch so should be good to go!

**Pull request merged**